### PR TITLE
Update: SearchGUI 4.0.7 and Xtandem Conda package dependency

### PR DESCRIPTION
**Pull request by CarlosHorro**
This update updates SearchGUI Conda package to SG 4.0.7.
Previous versions have problems when executing into containers due do prebuilt executables (specially, XTandem).
Now I'm trying to reuse XTandem Conda package, which should work but some small adjustments similar to the ones done by @bernt-matthias to [mpa-portable](https://github.com/bernt-matthias/bioconda-recipes/blob/ccde55f1e435d22e77210f716bcf4168df419165/recipes/mpa-portable/meta.yaml) .
However, after many attempts I'm having many problems for making it to work within containers:

Running 
`
bioconda-utils build --docker --mulled-test --git-range master --packages searchgui
`

I get

```

00:02:55 BIOCONDA INFO (OUT) Making absolute symlink bin/searchgui -> /opt/conda/conda-bld/searchgui_1608332332270/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/share/searchgui-4.0.6-1/searchgui relative
00:02:55 BIOCONDA INFO (OUT) Error: share/searchgui-4.0.6-1/resources/XTandem/linux/linux_64bit/tandem is a symlink to a path that may not exist after the build is completed (/opt/conda/conda-bld/searchgui_1608332332270/_build_env/bin/xtandem)
00:02:58 BIOCONDA ERROR COMMAND FAILED (exited with 1): docker run -t --net host --rm -v /tmp/tmplyhs8drk/build_script.bash:/opt/build_script.bash -v /tmp/miniconda/miniconda/conda-bld/:/opt/host-conda-bld -v /home/carlos/GitHub/carlos-bioconda-recipes/recipes/searchgui:/opt/recipe -e LC_ADDRESS=nb_NO.UTF-8 -e LC_NAME=nb_NO.UTF-8 -e LC_MONETARY=nb_NO.UTF-8 -e LC_PAPER=nb_NO.UTF-8 -e LANG=en_GB.UTF-8 -e LC_IDENTIFICATION=nb_NO.UTF-8 -e LC_TELEPHONE=nb_NO.UTF-8 -e LC_MEASUREMENT=nb_NO.UTF-8 -e LC_TIME=C -e LC_NUMERIC=nb_NO.UTF-8 -e HOST_USER_ID=1000 bioconda/bioconda-utils-build-env:latest /bin/bash /opt/build_script.bash

00:02:58 BIOCONDA ERROR BUILD FAILED recipes/searchgui
00:02:58 BIOCONDA INFO (COMMAND) conda build purge
00:03:00 BIOCONDA ERROR BUILD SUMMARY: of 1 recipes, 1 failed and 0 were skipped. Details of recipes and environments follow.
00:03:00 BIOCONDA ERROR BUILD SUMMARY: FAILED recipe recipes/searchgui


```

But, if I remove the ` build: - {{ compiler('c') }} ` (which has been always there) I get many warnings (from other prebuilt executables SG includes) but the building of the image and the tests are finished, although with internal errors:

```

00:27:09 BIOCONDA INFO (ERR) [Dec 19 00:27:09] INFO Invoke Task [test]
00:27:09 BIOCONDA INFO (ERR) [Dec 19 00:27:09] STEP Run image [quay.io/biocontainers/searchgui:4.0.6--1] with command [[searchgui eu.isas.searchgui.cmd.SearchCLI -id_params "Identification_Parameters_default.par" -fasta_file "fastacli_searchgui_tinydb1_concatenated_target_decoy.fasta" -spectrum_files "searchgui_tinyspectra1.mgf" -xtandem 1 -output_folder "." -output_option 0  -output_data 1]]
00:27:09 BIOCONDA INFO (ERR) [Dec 19 00:27:09] DEBU Creating container [step-c462cd727a]
00:27:09 BIOCONDA INFO (ERR) [Dec 19 00:27:09] DEBU Created container [b141234ee6d4 step-c462cd727a], starting it
00:27:09 BIOCONDA INFO (ERR) [Dec 19 00:27:09] DEBU Container [b141234ee6d4 step-c462cd727a] started, waiting for completion
00:27:09 BIOCONDA INFO (ERR) [Dec 19 00:27:09] SOUT Fri Dec 18 23:27:09 GMT 2020 An error occurred while running the command line. Please see the SearchGUI log file: /usr/local/share/searchgui-4.0.6-1/resources/SearchGUI.log
00:27:10 BIOCONDA INFO (ERR) [Dec 19 00:27:10] DEBU Container [b141234ee6d4 step-c462cd727a] completed with exit code [0] as expected
00:27:10 BIOCONDA INFO (ERR) [Dec 19 00:27:10] DEBU Container [b141234ee6d4 step-c462cd727a] removed
.00:27:14 BIOCONDA INFO (OUT) ....

00:27:14 BIOCONDA INFO TEST SUCCESS recipes/searchgui
00:27:14 BIOCONDA INFO (COMMAND) conda build purge
00:27:15 BIOCONDA INFO BUILD SUMMARY: successfully built 1 of 1 recipes


```
Please note that I've added one test to the recipe with some basic files in order to easily test the execution of SG with XTandem.

Greetings,
Carlos
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/peptide-shaker/meta.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ build:
   number: 0
   #noarch: generic
 
+
```
Comment (bernt-matthias) [inline recipes/peptide-shaker/meta.yaml:13]: I think you will need to undo this line insertion

**Comment by bernt-matthias** [comment]
Can you cleanup the commit history? Looks like there are plenty old commits that are already merged. Maybe just open a new PR with a single commit?

File: recipes/searchgui/meta.yaml [hunk 2]
```diff
@@ -15,26 +15,15 @@ source:
...
-  #run_exports:
-  #  strong:
-  #    - zlib
-  #    - libstdcxx-ng
-  #    - expat
+  
   build:
     - {{ compiler('c') }}
-    #- {{ compiler('cxx') }}
+  host:
```
Comment (bernt-matthias) [inline recipes/searchgui/meta.yaml:21]: Thanks @dpryan79 

Ah, I see. Of course we can set the link without having xtandem installed. Its then just a stale link during building. But then it should also not be required in the host requirements, or? 

But tests still don't seem to run...

**Comment by CarlosHorro** [comment]
Thanks for the help. I've received an email about this failure with a weird output message:


```

#!/bin/bash -eo pipefail
bioconda-utils lint recipes config.yml \
--loglevel debug --full-report \
--git-range master HEAD

16:23:35 BIOCONDA DEBUG get_recipes(recipes, package='['*']'): *
16:23:35 BIOCONDA INFO Considering total of 7588 recipes.
16:23:35 GIT DEBUG Popen(['git', 'remote', 'get-url', '--all', 'origin'], cwd=/home/circleci/project, universal_newlines=False, shell=None, istream=None)
16:23:35 BIOCONDA ERROR Dropping into debugger
Traceback (most recent call last):
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 71, in wrapper
    func(*args, **kwargs)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 345, in do_lint
    recipes = get_recipes(config, recipe_folder, packages, git_range)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 168, in get_recipes
    changed_recipes = get_recipes_to_build(git_range, recipe_folder)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 151, in get_recipes_to_build
    repo = BiocondaRepo(recipe_folder)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/githandler.py", line 503, in __init__
    super().__init__(repo, dry_run, home, fork, allow_dirty)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/githandler.py", line 83, in __init__
    self.home_remote = self.get_remote(home)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/githandler.py", line 143, in get_remote
    raise KeyError(f"No remote matching '{desc}' found")
KeyError: "No remote matching 'bioconda/bioconda-recipes' found"
Traceback (most recent call last):
  File "/home/circleci/project/miniconda/bin/bioconda-utils", line 8, in <module>
    sys.exit(main())
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 971, in main
    bioconductor_skeleton, clean_cran_skeleton, autobump, bot
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/argh/dispatching.py", line 328, in dispatch_commands
    dispatch(parser, *args, **kwargs)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/argh/dispatching.py", line 174, in dispatch
    for line in lines:
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/argh/dispatching.py", line 277, in _execute_command
    for line in result:
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/argh/dispatching.py", line 260, in _call
    result = function(*positional, **keywords)
  File "<boltons.funcutils.FunctionBuilder-3>", line 8, in do_lint
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 130, in wrapper
    func(*args, **kwargs)
  File "<boltons.funcutils.FunctionBuilder-2>", line 8, in do_lint
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 59, in wrapper
    func(*args, **kwargs)
  File "<boltons.funcutils.FunctionBuilder-1>", line 8, in do_lint
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 71, in wrapper
    func(*args, **kwargs)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 345, in do_lint
    recipes = get_recipes(config, recipe_folder, packages, git_range)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 168, in get_recipes
    changed_recipes = get_recipes_to_build(git_range, recipe_folder)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/cli.py", line 151, in get_recipes_to_build
    repo = BiocondaRepo(recipe_folder)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/githandler.py", line 503, in __init__
    super().__init__(repo, dry_run, home, fork, allow_dirty)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/githandler.py", line 83, in __init__
    self.home_remote = self.get_remote(home)
  File "/home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/githandler.py", line 143, in get_remote
    raise KeyError(f"No remote matching '{desc}' found")
KeyError: "No remote matching 'bioconda/bioconda-recipes' found"


Exited with code exit status 1
CircleCI received exit code 1
```

File: recipes/peptide-shaker/meta.yaml (hunk 1 above)
Comment (CarlosHorro) [reply recipes/peptide-shaker/meta.yaml]: Do extra empty lines modify the linting or the building of a Conda package?

**Comment by dpryan79** [comment]
@CarlosHorro Please disable circleci on your fork.

File: recipes/peptide-shaker/meta.yaml (hunk 1 above)
Comment (bernt-matthias) [reply recipes/peptide-shaker/meta.yaml]: I would expect that any modification will make bioconda think its an update and then the linter will complain about a missing bump.

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**